src/stores/modalContext.tsx

```diff
@@ -0,0 +1,235 @@
+import React, { createContext, useContext, useRef, useState } from 'react'
+import { ModalRef } from '@trussworks/react-uswds'
+import { useAnalytics } from 'stores/analyticsContext'
+import { useAddBookmarkMutation } from 'operations/portal/mutations/addBookmark.g'
+import { useRemoveCollectionMutation } from 'operations/portal/mutations/removeCollection.g'
+import { useRemoveWidgetMutation } from 'operations/portal/mutations/removeWidget.g'
+import { useRemoveBookmarkMutation } from 'operations/portal/mutations/removeBookmark.g'
+import { useEditBookmarkMutation } from 'operations/portal/mutations/editBookmark.g'
+import { Widget, Bookmark as BookmarkType } from 'types/index'
+
+// Try to clean this up
```

Comment:
Q for this file overall: What would it look like if we were to pull out all the specific-modal instance code in this file (e.g. remove bookmark modal, edit bookmark modal, etc), and pass it in to `useModalContext` instead from the relevant components?

i might be missing something, i am just imagining that the more modals we add, the more verbose this file will get. My brain keeps thinking of `modalContext.tsx` as the most abstract storage space for any modal, depending on what is passed in.

Reply:
Agreed that there is a lot of potential for `modalContext` to become increasingly complex, and I've been thinking about a couple of ways I could tighten it up and make it cleaner. However, if I'm understanding your question correctly, we can't pass those values in because any component that would call to `useModalContext` would be a child of the `<ModalProvider />` in `_app.tsx`. And in React, we can't pass data up the component tree, we can only utilize what is either in a parent component, or what's in the current component.

Reply:
Wait, maybe ignore my last, pre-coffee comment. I think I understand now. Are you asking if we could have a callback function in the context, and then pass the handler (remove bookmark, etc.) from the component into the context that way? I did try some variation of that and was having trouble getting it working. I think something of that nature could work, but not sure how to structure it. Maybe we could pair on it, if you have some ideas?

Reply:
yeah something like that! i'd definitely need to pair on it and think through. i remember now you mentioning trying something like that before. if not in this PR, a follow-up story? what do you think @gidjin

Reply:
Yeah I think putting the logic for each modal closer to its location and passing it in as a callback to the generic modal is where I would want to go. This would be good discussion for our Monday eng sync, since I too wouldn't might time work through it more together.

That said, I feel this is better than what we had before and I count this as a step in the right direction. So I'd be fine merging this in as is and creating a follow up tech liability story to pair on improving this work to refactor the modalContext further.

Reply:
Agreed. As I make updates to remove all of the comments I left, I'll add a new one as a to do at the top of `modalContext`. Thanks to both of you for the feedback!